assets/js/blocks-registry/payment-methods/assertions.js

```diff
@@ -0,0 +1,41 @@
+export const assertValidPaymentMethodComponent = ( paymentMethod ) => {
+	// @todo detect if functional component (not render prop)
+	if ( typeof paymentMethod !== 'function' ) {
+		throw new Error(
+			'The registered payment method must be a functional component'
+		);
+	}
+};
+
+export const assertValidPaymentMethod = ( paymentMethod ) => {
+	// paymentMethods are expected to have 4 properties, tab, content, name, ariaLabel.
```

Comment:
Apart from those 4 props, I can think of some more that may be needed:
- `extraFields`: Some gateways require extra fields on the checkout, such as `phone`. Right now, because the shipping/billind details must be rendered server-side, those extra fields are added if the gateway is active, regardless if the customer is trying to pay with said gateway. Maybe we can do better in this iteration. Altough I don't see how would that be possible if the checkout has several steps :(
- `submitButtonLabel`: Some extensions change the checkout button text. For example, if you're buying a subscription, the label changes to `SIGN UP`. Not sure if this should be gateway-specific though.
- `capabilities`: For example, to declare that the gateway allows tokenization (to save the cc details for next purchases). Not sure if the UI for that (which would be just a checkbox) should be inside the gateway `cotent` or be more generic.

Reply:
These properties here are what the payment method defines when it registers itself with checkout.  However the payment method component will receive additional properties when it is rendered including `checkoutData` which could include values for all the fields along with totals, line items etc that the payment method needs (from the checkout).  See what happens with the [express payment method component](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1349/commits/0b430a2e3370a15bb2410702e2716f2066627c86#diff-a7ea89ea744820f0d5b05a196c624c4cR22-R25) and the [payment method component](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1349/commits/0b430a2e3370a15bb2410702e2716f2066627c86#diff-cea364bc272eb722c3373b4d6e828d14R69-R73).

I think the payment method should also be responsible for any tokenization as that's more a property of the payment method itself.